JustSaying.Models/MessageAttributeValue.cs

```diff
@@ -31,7 +32,7 @@ public class MessageAttributeValue
         /// data, or images.
         /// </para>
         /// </summary>
-        public byte[] BinaryValue { get; set; }
+        public IEnumerable<byte> BinaryValue { get; set; }

```

Comment:
`ICollection<T>` or `IList<T>` is much better as you can use `.Count` rather than needing to turn it into a more derived type. `IEnumerable<T>` as an output type (rather than an input) also can lead to bugs due to unmaterialized collections being passed around, causing enumeration multiple times, which is wasteful.

`IList<T>` would be the most compatible replacement as it has the larger surface area, and also retains the indexer, which `ICollection<T>` would not.

Reply:
Hm, we don't actually count it, that's the only use in `SnsTopicBase`. 
I favour `IReadOnlyCollection<T>` in other cases.

Reply:
`I*Collection<T>` or `I*List<T>` is preferable, IMHO, and still makes the rule happy.

I think the fix hints for this rule are poor, as I believe the real intention of the rule is not suggest against not using concrete types (`T[]`, `List<T>`, `Dictionary<K, V>`), rather than having to use `IEnumerable<T>` _specifically_.